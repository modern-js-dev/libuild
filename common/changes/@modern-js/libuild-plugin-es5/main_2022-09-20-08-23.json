{
  "changes": [
    {
      "packageName": "@modern-js/libuild-plugin-es5",
      "comment": "",
      "type": "none"
    }
  ],
  "packageName": "@modern-js/libuild-plugin-es5"
}