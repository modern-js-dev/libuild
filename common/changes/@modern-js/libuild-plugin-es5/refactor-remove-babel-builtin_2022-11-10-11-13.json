{
  "changes": [
    {
      "packageName": "@modern-js/libuild-plugin-es5",
      "comment": "deprecate es5InputPlugin and es5OutputPlugin, use es5Plugin powered by swc",
      "type": "none"
    }
  ],
  "packageName": "@modern-js/libuild-plugin-es5"
}