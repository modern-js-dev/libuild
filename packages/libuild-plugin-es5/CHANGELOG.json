{
  "name": "@modern-js/libuild-plugin-es5",
  "entries": [
    {
      "version": "0.3.0",
      "tag": "@modern-js/libuild-plugin-es5_v0.3.0",
      "date": "Thu, 03 Nov 2022 06:41:40 GMT",
      "comments": {
        "dependency": [
          {
            "comment": "Updating dependency \"@modern-js/libuild\" to `0.3.0`"
          }
        ]
      }
    },
    {
      "version": "0.1.4",
      "tag": "@modern-js/libuild-plugin-es5_v0.1.4",
      "date": "Thu, 13 Oct 2022 12:46:42 GMT",
      "comments": {
        "dependency": [
          {
            "comment": "Updating dependency \"@modern-js/libuild\" to `0.2.0`"
          }
        ]
      }
    },
    {
      "version": "0.1.3",
      "tag": "@modern-js/libuild-plugin-es5_v0.1.3",
      "date": "Wed, 12 Oct 2022 03:51:07 GMT",
      "comments": {
        "dependency": [
          {
            "comment": "Updating dependency \"@modern-js/libuild\" to `0.1.3`"
          }
        ]
      }
    },
    {
      "version": "0.1.2",
      "tag": "@modern-js/libuild-plugin-es5_v0.1.2",
      "date": "Mon, 26 Sep 2022 03:44:18 GMT",
      "comments": {
        "dependency": [
          {
            "comment": "Updating dependency \"@modern-js/libuild\" to `0.1.2`"
          }
        ]
      }
    },
    {
      "version": "0.1.1",
      "tag": "@modern-js/libuild-plugin-es5_v0.1.1",
      "date": "Tue, 20 Sep 2022 08:24:02 GMT",
      "comments": {
        "dependency": [
          {
            "comment": "Updating dependency \"@modern-js/libuild\" to `0.1.1`"
          }
        ]
      }
    }
  ]
}
