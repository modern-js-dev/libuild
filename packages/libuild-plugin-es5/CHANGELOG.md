# Change Log - @modern-js/libuild-plugin-es5

This log was last generated on Thu, 01 Dec 2022 08:58:38 GMT and should not be manually modified.

## 0.5.1
Thu, 01 Dec 2022 08:58:38 GMT

_Version update only_

## 0.5.0
Wed, 30 Nov 2022 09:08:41 GMT

_Version update only_

## 0.4.0
Mon, 14 Nov 2022 06:16:02 GMT

### Updates

- deprecate es5InputPlugin and es5OutputPlugin, use es5Plugin powered by swc

## 0.3.0
Thu, 03 Nov 2022 07:55:22 GMT

_Version update only_

## 0.1.4
Thu, 13 Oct 2022 12:46:42 GMT

_Version update only_

## 0.1.3
Wed, 12 Oct 2022 03:51:07 GMT

_Version update only_

## 0.1.2
Mon, 26 Sep 2022 03:44:18 GMT

_Version update only_

## 0.1.1
Tue, 20 Sep 2022 08:24:02 GMT

_Initial release_

