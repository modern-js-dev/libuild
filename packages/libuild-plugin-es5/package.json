{
  "name": "@modern-js/libuild-plugin-es5",
  "version": "0.4.0",
  "description": "A plugin for libuild to transform your code to es5",
  "keywords": [
    "libuild",
    "es5"
  ],
  "repository": "modern-js-dev/libuild",
  "main": "./dist/cjs/index.js",
  "module": "./dist/esm/index.js",
  "types": "./dist/index.d.ts",
  "scripts": {
    "dev": "libuild -w",
    "build": "tsc && libuild"
  },
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "typescript": "4.4.4",
    "@modern-js/eslint-config": "workspace:*",
    "@modern-js/libuild": "workspace:*",
    "@modern-js/libuild-utils": "workspace:*",
    "@modern-js/libuild-node-rig": "workspace:*"
    },
  "dependencies": {
    "@swc/core": "1.3.14"
  },
  "files": [
    "dist",
    "!dist/*.map"
  ]
}
