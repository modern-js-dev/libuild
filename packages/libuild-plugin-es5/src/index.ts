export { es5InputPlugin } from './input';
export { es5OutputPlugin } from './output';
