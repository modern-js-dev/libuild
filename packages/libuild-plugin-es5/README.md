# @modern-js/libuild-plugin-es5
[![npm version](https://badge.fury.io/js/@modern-js%2Flibuild-plugin-es5.svg)](https://www.npmjs.com/package/@modern-js/libuild-plugin-es5)

A plugin for libuild to transform your code to es5.

## Usage

```ts
// libuild.config.ts
import { defineConfig } from '@modern-js/libuild';
import { es5Plugin } from '@modern-js/libuild-plugin-es5';

export = defineConfig({
  plugins:[
    es5Plugin()
  ]
})
```
