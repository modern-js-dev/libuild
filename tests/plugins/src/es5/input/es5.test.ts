import { getLibuilderTest } from '@modern-js/libuild-test-toolkit';
import { es5InputPlugin } from '@modern-js/libuild-plugin-es5';

describe('fixture:es5', () => {
  it('es5-input', async () => {
    const bundler = await getLibuilderTest({
      root: __dirname,
      input: {
        ts: './index.ts',
        js: './index.js',
        tsx: './index.tsx',
        jsx: './index.jsx',
      },
      target: 'es5',
      plugins: [es5InputPlugin()],
    });
    await bundler.build();
    bundler.expectJSOutputMatchSnapshot();
  });
});
