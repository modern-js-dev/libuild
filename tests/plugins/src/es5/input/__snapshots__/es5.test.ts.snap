// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fixture:es5 es5-input 1`] = `
"// index.ts
var value = 1;
console.log(value);
"
`;

exports[`fixture:es5 es5-input 2`] = `
"// index.js
var value = 1;
console.log(value);
"
`;

exports[`fixture:es5 es5-input 3`] = `
"// index.tsx
var value = 1;
console.log(value);
"
`;

exports[`fixture:es5 es5-input 4`] = `
"// index.jsx
var value = /* @__PURE__ */ React.createElement(\\"div\\", null);
console.log(value);
"
`;
